soc/arm/atmel_sam/same70/Kconfig.series

```diff
@@ -10,7 +10,8 @@ config SOC_SERIES_SAME70
 	select SOC_FAMILY_SAM
 	select CPU_HAS_FPU_DOUBLE_PRECISION
 	select CPU_CORTEX_M_HAS_DWT
-	select CPU_CORTEX_M_HAS_CACHE
+	select DCACHE
+	select ICACHE
```

Comment:
Do we still need that? This select `CPU_CORTEX_M7`, which now selects `CPU_HAS_DCACHE`and `CPU_HAS_ICACHE`. `DCACHE`and `ICACHE` are enabled by default when `CPU_HAS_DCACHE`and `CPU_HAS_ICACHE` are enabled.

Ditto for all the ARM SoCs.

Reply:
> DCACHEand ICACHE are enabled by default when CPU_HAS_DCACHEand CPU_HAS_ICACHE are enabled.

This is true only for ARM64 but not for all the other architectures (including ARM).

I can add that for `CPU_CORTEX_M7` as well but for now ARM has a finer granularity when it comes to caches.

Reply:
should not select `DCACHE` and `ICACHE` here -- that effectively makes these caches compulsory for this SoC (users cannot de-select them).

these should be `default y` in the SoC defconfig.

Reply:
Actually not even that is needed because `DCACHE` has a `default y` on `CPU_HAS_DCACHE` (same for icache).